wgpu-core/src/device/mod.rs

```diff
@@ -192,21 +192,14 @@ fn map_buffer<A: hal::Api>(
     let zero_init_needs_flush_now = mapping.is_coherent && buffer.sync_mapped_writes.is_none(); // No need to flush if it is flushed later anyways.
     for uninitialized_range in buffer.initialization_status.drain(offset..(size + offset)) {
         let num_bytes = uninitialized_range.end - uninitialized_range.start;
-        unsafe {
-            ptr::write_bytes(
-                mapping
-                    .ptr
-                    .as_ptr()
-                    .offset(uninitialized_range.start as isize),
-                0,
-                num_bytes as usize,
-            )
-        };
+        // The mapping's pointer is already offset, however we track the uninitialized range relative to the buffer's start.
+        let start = (uninitialized_range.start - offset) as isize;
+        unsafe { ptr::write_bytes(mapping.ptr.as_ptr().offset(start), 0, num_bytes as usize) };
```

Comment:
Could set just remove the unsafe here? I don't think this is giving us meaningful performance boosts.

Reply:
Good point, I updated the patch. We can't entirely get rid of the unsafe (since we strat from an unsafe pointer), but the part unsafe piece is now pretty trivial and the part that writes the range is now safe and bounds checked. I also checked with godbolt that `slice.fill(0)` desugars into an memset (at least with u8 slices).